server/service/appconfig.go

```diff
@@ -326,6 +326,16 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 		err = fleet.NewUserMessageError(err, http.StatusBadRequest)
 		return nil, ctxerr.Wrap(ctx, err)
 	}
+
+	// EnableDiskEncryption is an optjson.Bool field in order to support the
+	// legacy field under "mdm.macos_settings". If the field provided to the
+	// PATCH endpoint is set but invalid (that is, "enable_disk_encryption":
+	// null) and no legacy field overwrites it, leave it unchanged (as if not
+	// provided).
+	if appConfig.MDM.EnableDiskEncryption.Set && !appConfig.MDM.EnableDiskEncryption.Valid {
+		appConfig.MDM.EnableDiskEncryption = oldAppConfig.MDM.EnableDiskEncryption
+	}
```

Comment:
sanity checking myself: there's a similar check here

https://github.com/fleetdm/fleet/blob/57c3ec2ca5e49c899cba3ac6476bfbd9c6ab60e0/server/service/appconfig.go#L370-L377

but they can't be combined, right?

Reply:
Oh good catch, I hadn't seen this! I can definitely move the code close together, but they are complimentary (the one I added is the other way around, I think, i.e. if the provided value was `null` use the old one, while this sets the new one if one was provided). Thinking about it, is this check really necessary? The new app config is basically what was provided in the JSON, and would've overwritten the `appConfig` content anyway when merging both together? I'll leave it just to be safe, but I'm wondering.

Reply:
thanks! I hate myself for not adding a more accurate comment there. I do remember I had to do this because something in the unmarshaller wasn't working here:

https://github.com/fleetdm/fleet/blob/57c3ec2ca5e49c899cba3ac6476bfbd9c6ab60e0/server/service/appconfig.go#L296-L302

Reply:
:+1: sounds good, yeah I'll keep it there for sure, this whole ModifyAppConfig is so tricky.